src/writers/post_operator/cargo_dependency_writer.rs

```diff
@@ -38,5 +47,25 @@ pub fn write_cargo_toml_file(base_path_str: &String) -> std::io::Result<()> {
     let cargo_deps = get_cargo_dependency_string();
     file.write_all(cargo_deps.as_bytes())?;
 
+    // write the toml flag to file
+    file.write_all(String::from("\n[other]\nwritten = true").as_bytes())?;
+
     Ok(())
 }
+
+/// Returns `Ok(true)` if the file has a `written` flag set to `true`, else
+/// either returns false or returns an error if file is unnaccessible.
+fn check_written_already(path_str: &PathBuf) -> std::io::Result<bool> {
+    let contents = std::fs::read_to_string(path_str).expect("Unable to read file");
+    let parsed_toml: toml::Value = toml::from_str(&contents)?;
+
+    // get the last bool value
+    if let Some(other) = parsed_toml.get("other") {
+        if let Some(written) = other.get("written") {
+            if let Some(val) = written.as_bool() {
+                return Ok(val);
+            }
+        }
+    }
+    return Ok(false);
+}
```

Comment:
Technically this isn't the most absolutely airtight way of ensuring the file is only written to once, but it works well enough. The only way it falls apart is if the user edits it/removes the last line. Maybe we should add a comment to NOT edit it? In any case, it works.

Reply:
Why do we need this at all?

Reply:
when `cargo new` or `cargo init` is run, it auto-gens a `Cargo.toml` file that has package data that `cargo run` relies on as well as the header for writing dependencies (e.g. actix).

In order for our generated code to run and compile we need to make sure the required deps. are met in the form of writing them to the `Cargo.toml` file.

The problem comes in with the duplicate code: we can't really have a stateful application that knows when commands have been run.

Every time the user makes a change to their input toml file and wants to re-generate their code they're gonna have to run the init command which basically needs to ensure that the underlying aut-generated `Cargo.toml` file has the deps.

With this change, the write is guaranteed to only happen once, so long as the user doesn't mess with the generated files.

Reply:
Makes sense, I forgot users may generate more than once.